website/project/tasks.py

```diff
@@ -34,51 +35,76 @@ def on_node_updated(node_id, user_id, first_save, saved_fields, request_headers=
         update_share(node)
 
 def update_share(node):
+    # Wrapper that ensures share_url and token exist
     if settings.SHARE_URL:
         if not settings.SHARE_API_TOKEN:
-            return logger.warning('SHARE_API_TOKEN not set. Could not send %s to SHARE.'.format(node))
-        if node.is_registration:
-            on_registration_updated(node)
+            return logger.warning('SHARE_API_TOKEN not set. Could not send "{}" to SHARE.'.format(node._id))
+        _update_share(node)
+
+def _update_share(node):
+    # Any modifications to this function may need to change _async_update_share
+    data = serialize_share_data(node)
+    resp = send_share_data(data)
+    try:
+        resp.raise_for_status()
+    except Exception:
+        if resp.status_code >= 500:
+            _async_update_share.delay(node._id)
         else:
-            update_node_share(node)
+            send_desk_share_error(node, resp, 0)
 
-def update_node_share(node):
-    resp = requests.post('{}api/normalizeddata/'.format(settings.SHARE_URL), json={
-        'data': {
-            'type': 'NormalizedData',
-            'attributes': {
-                'tasks': [],
-                'raw': None,
-                'data': {'@graph': [{
-                    '@id': '_:123',
-                    '@type': 'workidentifier',
-                    'creative_work': {'@id': '_:789', '@type': 'project'},
-                    'uri': '{}{}/'.format(settings.DOMAIN, node._id),
-                }, {
-                    '@id': '_:789',
-                    '@type': 'project',
-                    'is_deleted': not node.is_public or node.is_deleted or node.is_spammy,
-                }]}
-            }
-        }
-    }, headers={'Authorization': 'Bearer {}'.format(settings.SHARE_API_TOKEN), 'Content-Type': 'application/vnd.api+json'})
+@celery_app.task(bind=True, max_retries=4, acks_late=True)
+def _async_update_share(self, node_id):
+    # Any modifications to this function may need to change _async_update_share
```

Comment:
I went through and renamed all of the _update_share functions to be _update_node_share to be less ambiguous. Pushing in just a moment. Should clarify this.